src/main.c

```diff
@@ -288,6 +291,46 @@ static uint32_t read_encoder_right(void)
 
 
 /**
+ * @brief Setup for adc: Four injected channels on scan mode for ADC1.
+ * - Initialize channel_sequence structure to map physical channels vs
+ * sw injected channels.
+ * - Power off the ADC to be sure that doesn't run during configuration.
+ * - Enable scan mode with single conversion mode triggered by software.
+ * - Configure the alignment (right) and the sample time (28.5 cycles of ADC
+ * clock)
+ * - Set injected sequence with channel_sequence structure
+ * - Power on the ADC and wait for ADC starting up (about 3 ms, See adc_power_on
+ * function from libopencm3)
+ * - Calibrate the ADC
+ *
+ * @see Reference manual (RM0008) "Analog-to-digital converter" and in
+ * particular "Scan mode" section.
+ */
+static void setup_adc(void)
+{
+	int i;
+
+	uint8_t channel_sequence[4] = {ADC_CHANNEL4, ADC_CHANNEL5, ADC_CHANNEL6,
+	ADC_CHANNEL7};
+
+	adc_power_off(ADC1);
+	adc_enable_scan_mode(ADC1);
+	adc_set_single_conversion_mode(ADC1);
+	adc_enable_external_trigger_injected(ADC1, ADC_CR2_JEXTSEL_JSWSTART);
+	adc_set_right_aligned(ADC1);
+	adc_set_sample_time_on_all_channels(ADC1, ADC_SMPR_SMP_28DOT5CYC);
+	adc_set_injected_sequence(ADC1,
+		sizeof(channel_sequence) / sizeof(channel_sequence[0]),
+		channel_sequence);
+	adc_power_on(ADC1);
+	for (i = 0; i < 800000; i++)
+		__asm__("nop");
```

Comment:
Not very important now, but if in the future we do not know what to do...

https://github.com/Theseus/theseus/issues/59

:joy:

Reply:
good